dask/dataframe/io.py

```diff
@@ -462,14 +463,18 @@ def to_hdf(df, path_or_buf, key, mode='a', append=False, complevel=0,
                  "must preserve the order of its input")
 
     # handle lock default based on wether we're writing to a single entity
+    _actual_get = get or _globals.get('get') or get_sync
```

Comment:
I didn't understand it initially but using `get_sync` on sequential writes is important to keep current usage. Without doing so users will experience performance degradation.
Removing that in the future seems cleaner, we'll see.

I will now commit your proposed policy, sorry it took me some time to fully understand.